monitoring/timedrift_test.go

```diff
@@ -0,0 +1,189 @@
+				node3.Name: s.clock.Now().Add(500 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.successProbe(node2, 100*time.Millisecond),
+				s.c.failureProbe(node3, 500*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to node-2 is high",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(-time.Second),
+				node3.Name: s.clock.Now().Add(100 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.failureProbe(node2, -time.Second),
+				s.c.successProbe(node3, 100*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to both nodes is high",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(-time.Second),
+				node3.Name: s.clock.Now().Add(time.Second),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.failureProbe(node2, -time.Second),
+				s.c.failureProbe(node3, time.Second),
+			},
+		},
+	}
+	for _, test := range tests {
+		clients, err := ttlmap.New(10)
+		c.Assert(err, check.IsNil)
+		for nodeName, nodeTime := range test.Times {
+			clients.Set(
+				nodes[nodeName].Addr.String(),
+				newMockedTimeAgentClient(nodeTime),
+				time.Hour)
+		}
+		checker := &timeDriftChecker{
+			TimeDriftCheckerConfig: TimeDriftCheckerConfig{
+				Clock: s.clock,
+			},
+			FieldLogger: logrus.WithField(trace.Component, "test"),
+			clients:     clients,
+			serfClient:  agent.NewMockSerfClient(test.Nodes, nil),
+			name:        node1.Name,
+			self:        node1,
+		}
+		var probes health.Probes
+		checker.Check(context.TODO(), &probes)
+		c.Assert(probes.GetProbes(), check.DeepEquals, test.Result,
+			check.Commentf(test.Comment))
+	}
+}
+
+type mockedTimeAgentClient struct {
+	agent.Client
+	time time.Time
+}
+
+func newMockedTimeAgentClient(time time.Time) *mockedTimeAgentClient {
+	return &mockedTimeAgentClient{time: time}
+}
+
+func (a *mockedTimeAgentClient) Time(ctx context.Context, req *pb.TimeRequest) (*pb.TimeResponse, error) {
+	return &pb.TimeResponse{
+		Timestamp: pb.NewTimeToProto(a.time),
+	}, nil
+}
+
+var (
+	node1 serf.Member = serf.Member{
+		Name:   "node-1",
+		Status: agentpb.MemberStatus_Alive.String(),
+		Addr:   net.IPv4(192, 168, 0, 1),
+	}
+	node2 serf.Member = serf.Member{
+		Name:   "node-2",
+		Status: agentpb.MemberStatus_Alive.String(),
+		Addr:   net.IPv4(192, 168, 0, 2),
+	}
+	node3 serf.Member = serf.Member{
+		Name:   "node-3",
+		Status: agentpb.MemberStatus_Alive.String(),
+		Addr:   net.IPv4(192, 168, 0, 3),
+	}
+	nodes map[string]serf.Member = map[string]serf.Member{
```

Comment:
`nodes` is not used beyond the bit that initializes `clients` - this belongs, for instance, in `newClients` instead (where the clients initialization is abstracted)

Reply:
Don't see anything wrong with keeping it here.